test/listen.test.js

```diff
@@ -156,6 +156,20 @@ test('double listen errors callback with (err, address)', t => {
   })
 })
 
+test('nonlocalhost double listen errors callback with (err, address)', t => {
+  t.plan(4)
+  const fastify = Fastify()
+  t.teardown(fastify.close.bind(fastify))
+  fastify.listen({ host: '::1', port: 0 }, (err, address) => {
+    t.equal(address, `http://${'[::1]'}:${fastify.server.address().port}`)
+    t.error(err)
+    fastify.listen({ host: '::1', port: fastify.server.address().port }, (err2, address2) => {
```

Comment:
nitpick:

Usually do the t.error call as the first assertion. fail fast, fail early.

```suggestion
  fastify.listen({ host: '::1', port: 0 }, (err, address) => {
    t.error(err)
    t.equal(address, `http://${'[::1]'}:${fastify.server.address().port}`)

    fastify.listen({ host: '::1', port: fastify.server.address().port }, (err2, address2) => {
```